rts/Sim/Misc/Wind.cpp

```diff
@@ -120,7 +121,14 @@ void EnvResourceHandler::Update()
 		curWindDir = curWindVec;
 		curWindVec = curWindDir * (curWindStrength = std::clamp(curWindStrength, minWindStrength, maxWindStrength));
 
-		for (const int unitID: newGeneratorIDs) {
+		if (const auto& wcrp = modInfo.windChangeReportPeriod; wcrp > 0 && gs->frameNum % wcrp == 0) {
+			// update generators every modInfo.windChangeReportPeriod frames
+			for (auto unitID : allGeneratorIDs) {
+				(unitHandler.GetUnit(unitID))->UpdateWind(curWindDir.x, curWindDir.z, curWindStrength);
+			}
+		}
+
+		for (auto unitID: newGeneratorIDs) {
```

Comment:
Could be in an `else` clause, since otherwise generators made on an update frame get two events.